### ISSUE TITLE
File extension in new lighttable overlaps with thumbnail

### ISSUE THREAD

**Issue by Mark-64** [darktable-org/darktable#4849]
In the new lighttable, with the following overlay modes
- permanent overlays
- permanent extended overlays
- permanent overlays extended on mouse hover

the file extension indicator in the upper left of the thumbnails overlaps with the image (it goes under).
This happens only for certain zoom levels and it also depends on the image aspect ratio.
See the example below:

![Cattura_w5](https://user-images.githubusercontent.com/43290988/80514042-41bdd300-8980-11ea-91e5-dfe3a961481a.JPG)

Note that the "altered" and "group" icons never overlap, they are scaled correctly.
Perhaps it could be enough to align the file extension indicator to the very top of the thumbnail, just like the icons.

Tested with DT 3.1 git master on Windows 10 and Ubuntu 19.10, HiDPI display

**Comment by Nilvus** [comment]
@Mark-64: yes, I've seen that when working on CSS. It's just indeed on overlays modes you precise only. That don't bother me so I don't post an issue. Said that, it can't be change on CSS. From what I see, the only way would be to change Gtk code and move the extension name the box/widget above, so on same line as group/audio showed on the top right. And that's why altered/group icons are never overlapped.

And from what I see on how thumbnails are made with Gtk Inspector tool, that could move other things so not sure it will be easily done.

@AlicVB: what do you think about that?

**Comment by AlicVB** [comment]
yes, that's "by design". the new overlays mode just show that more obviously. Previously, if you try with a square image, for example,you'll see that.

**Comment by Mark-64** [comment]
In thumbnail.c, line 918, changing top margin for the extension label from 
`gtk_widget_set_margin_top(thumb->w_ext, 0.045 * width);`
to
`gtk_widget_set_margin_top(thumb->w_ext, 0.5 * r1);`
the label will be aligned with the altered / group / audio icons and it will never be cut.

![Screenshot from 2020-04-28 22-18-12](https://user-images.githubusercontent.com/43290988/80534074-f2d36600-899e-11ea-8e93-8464da57d8bf.png)

@AlicVB @Nilvus, it doesn't look bad to me, what do you think ?

**Comment by Nilvus** [comment]
That's just breaks also the goal of this new overlay mode: use more space for the image. See margins that adds on left, right and bottom in regards of your first screenshot with the "issue".

**Comment by AlicVB** [comment]
Well, I'm not a specialist in design, but the initial idea of the extension text is more to act as a background incrustation than an overlay. That explain lot of things : 
- it's placed under the image
- it can be vertical or horizontal, depending of the orientation of the image
- it's shown independently of the overlays mode

That said, I'm not against changing margins (but we'll need to change also the left one) We just need to ensure it's ok for different ratios/orientation/overlays

**Comment by AlicVB** [comment]
> That's just breaks also the goal of this new overlay mode: use more space for the image. See margins that adds on left, right and bottom in regards of your first screenshot with the "issue".

if he has just changed the mentioned line, it shouldn't have touch the rest. I think some other settings have changed (maybe a different thumb size)

**Comment by Mark-64** [comment]
I realize I posted thumbnails with different setting, so it wasn't clear, my bad.
Here is the thumbnail without the patch:
![Cattura28](https://user-images.githubusercontent.com/43290988/80643407-54133c00-8a68-11ea-98b7-c7894a885a76.JPG)

and here is with the patch

![Cattura29](https://user-images.githubusercontent.com/43290988/80643449-62615800-8a68-11ea-987e-a0dc10d4e48c.JPG)

There are no side effects, only the label is moved.
To me the upper margin is `0.5 * r1` already, because the top icons have that margin. So we just align the text with the icons and the result is pleasing at least to my eye.
Also, the text is resized together with the icons, so it will never be cut, because the icons are not cut.
When coming to the left margin, I would leave that as it is, moving the label too close to the left edge would not look good IMO.

I have tested with several zoom levels, image aspect ratios and orientations, it seems to work in all cases.

I will do a PR, so people can do other tests.
